packages/backend-core/src/util/jwt.ts

```diff
@@ -17,8 +17,14 @@ export function isExpired(
   }
 }
 
-export function checkClaims(claims: JWTPayload) {
+export function checkClaims(claims: JWTPayload, authorizedParties?: string[]) {
   if (!claims.iss || !claims.iss.startsWith('https://clerk')) {
     throw new Error(`Issuer is invalid: ${claims.iss}`);
   }
+
+  if (claims.azp && authorizedParties && authorizedParties.length > 0) {
```

Comment:
🙃 
Except if we want to be really strict, could we maybe reduce the check in this:
```ts
  if (!authorizedParties?.includes(claims.azp as string)) {
    throw new Error(`Authorized party is invalid: ${claims.azp}`);
  }
```